fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -602,10 +585,16 @@ protected void replaceResourceWithStream(final FedoraResource resource,
     }
 
     protected void patchResourcewithSparql(final FedoraResource resource,
-                                           final String requestBody,
-                                           final RdfStream resourceTriples)
-            throws MalformedRdfException, AccessDeniedException {
-        resource.updateProperties(translator(), requestBody, resourceTriples);
+            final String requestBody,
+            final RdfStream resourceTriples)
+                    throws MalformedRdfException, AccessDeniedException {
+        if (resource instanceof NonRdfSourceDescription) {
+            // update the description instead
```

Comment:
I think there are a couple of different priorities here, and different people care about different things, leading to some of this always-shifting discussion.  Some of the priorities are:

1. Full-featured descriptions of binaries (e.g., Skolemized bnodes, hash URI resources, child containers, etc.)
2. Binary descriptions tightly bound to the binary (e.g., deleted when the binary is deleted).
3. Request URI == RDF subject URI of the triples returned

My understanding was that the way we were going to handle these competing priorities was to keep fcr:metadata, make sure all binary properties used it as the RDF subject, and then add container/bnode/hashURI features to it.